code/modules/mapping/map_template.dm

```diff
@@ -65,14 +72,11 @@
 	// first or not.  Its defined In Initialize yet its run first in templates
 	// BEFORE so... hummm
 	SSmapping.reg_in_areas_in_z(areas)
-	// We have to do this hack here because its the ONLY place we can get the
-	// meta data from the template so we can properly set up the area
-	SSnetworks.assign_areas_root_ids(areas, template)
-	// If the world is starting up stop here and the world will do the rest
+	SSnetworks.assign_areas_root_ids(areas, src)
 	if(!SSatoms.initialized)
 		return
 
-	SSatoms.InitializeAtoms(areas + turfs + atoms)
+	SSatoms.InitializeAtoms(areas + turfs + atoms, returns_created_atoms ? created_atoms : null)
```

Comment:
This should register to the atom's destruction not to cause hard deletions.

Reply:
turns out the created_atoms list var in the map template and the spawned list var in the holodeck computer are both references to the same list, so whenever something in the list gets destroyed without changing programs its handled automatically by the holodeck computer